apps/vs-code-react/src/webviewCommunication.tsx

```diff
@@ -1,46 +1,153 @@
-        dispatch(updateAccessToken(message.data.accessToken));
-        break;
-      case ExtensionCommand.update_export_path:
-        dispatch(updateTargetDirectory(message.data));
-        break;
-      case ExtensionCommand.add_status:
-        dispatch(addStatus(message.data));
+    if (message.command === ExtensionCommand.initialize_frame) {
+      dispatch(initialize(message.data.project));
+    }
+
+    switch (projectState?.project) {
+      case ProjectName.designer:
+        switch (message.command) {
+          case ExtensionCommand.initialize_frame:
+            dispatch(initializeDesigner(message.data));
+            break;
+          case ExtensionCommand.receiveCallback:
+            dispatch(updateCallbackUrl(message.data));
+            break;
+          case ExtensionCommand.completeFileSystemConnection:
+            dispatch(updateFileSystemConnection(message.data));
+            break;
+          case ExtensionCommand.update_panel_metadata:
+            dispatch(updatePanelMetadata(message.data));
+            break;
+          default:
+            throw new Error('Unknown post message received');
+        }
         break;
-      case ExtensionCommand.set_final_status:
-        dispatch(setFinalStatus(message.data));
+      case ProjectName.dataMapper:
+        switch (message.command) {
+          case ExtensionCommand.setRuntimePort:
+            dispatch(changeRuntimePort(message.data));
+            break;
+          case ExtensionCommand.fetchSchema:
+            if (message.data.type === SchemaType.Source) {
+              dispatch(changeSourceSchemaFilename(message.data.fileName));
+            } else {
+              dispatch(changeTargetSchemaFilename(message.data.fileName));
+            }
+            break;
+          case ExtensionCommand.loadDataMap:
+            // NOTE: DataMapDataProvider ensures the functions and schemas are loaded before loading the mapDefinition connections
+            dispatch(changeSourceSchemaFilename(message.data.sourceSchemaFileName));
+            dispatch(changeTargetSchemaFilename(message.data.targetSchemaFileName));
+            dispatch(changeMapDefinition(message.data.mapDefinition));
+            dispatch(changeDataMapMetadata(message.data.metadata));
+            break;
+          case ExtensionCommand.showAvailableSchemas:
+            dispatch(changeSchemaList(message.data));
+            break;
+          case ExtensionCommand.getAvailableCustomXsltPaths:
+            dispatch(changeCustomXsltPathList(message.data));
+            break;
+          case ExtensionCommand.setXsltData:
+            dispatch(changeXsltFilename(message.data.filename));
+            dispatch(changeXsltContent(message.data.fileContents));
+            break;
+          case ExtensionCommand.getConfigurationSetting:
+            dispatch(changeUseExpandedFunctionCards(message.data));
+            break;
+          default:
+            console.warn(`Unexpected message received:`);
```

Comment:
Is there a reason on why it is only console logging the error and not throwing an error?

Reply:
I was unsure on the reason on why the original code was console logging instead of throwing error. I'll change that to reflect throwing error for other cases